### PR TITLE
add api.md and ts.md for perp

### PR DESCRIPTION
**Pull request by samweinberg23**
- for https://github.com/dydxprotocol/solo/issues/442

### CODE REVIEW

File: docs/perpetual-typescript.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  {
+    defaultAccount: '0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1', // Optional - but needed if using Infura
+    accounts: [
+      {
+        address: '0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1', // Optional
+        privateKey: '0x4f3edf983ac636a65a842ce7c78d9aa706d3b113bce9c46f30d7d21715b23b1d',
+      },
+    ], // Optional: loading in an account for signing transactions
+  }, // Optional
+);
```
Comment (BrendanChou) [inline docs/perpetual-typescript.md:55]: removed this option in perpetual. Either way they should pass in a valid provider object

File: docs/perpetual-typescript.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+```
+
+### Types
+You can import types from Perpetual as:
+
+```javascript
+import {
+  ProxyType,
+  AmountDenomination,
+  AmountReference,
```
Comment (BrendanChou) [inline docs/perpetual-typescript.md:132]: these three are not in the perpetual, maybe put `Order` and `SignedOrder` as examples instead

File: docs/perpetual-api.md [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+```
+
+Note: To get any account's collateralization, simply take `sumSupplyUsdValue / sumBorrowUsdValue`.
+The minimum collateralization where liquidation occurs on the protocol using this formula is 1.15.
+
+Query Params:
+
+|Field Name|Description|
+|----------|-----------|
+|maker|string|The Ethereum address of the Maker.|
```
Comment (BrendanChou) [inline docs/perpetual-api.md:259]: delete this

File: docs/perpetual-api.md [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+
+|Field Name|JSON type|Description|
+|----------|---------|-----------|
+|order|Object|A valid signed V2 order JSON object|
+|fillOrKill|boolean|Whether the order should be canceled if it cannot be immediately filled|
+|postOnly|boolean|Whether the order should be canceled if it would be immediately filled|
+|triggerPrice|(Optional)The price at which the order will go to market. Must be greater than triggerPrice in the order|
+|cancelId|string|(Optional)Order id for the order that is being canceled and replaced|
+|clientId|string|(Optional)An arbitrary string guaranteed to be unique for each makerAccountOwner. Will be returned alongside the order in subsequent requests.|
+|setExpirationOnFill|boolean|(Optional)Expiration field for order will be applied upon the order filling.|
```
Comment (BrendanChou) [inline docs/perpetual-api.md:102]: delete this

File: docs/perpetual-typescript.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  accountOwner: '0x52bc44d5378309ee2abf1539bf71de1b7d7be3b5',
+});
+```
+
+#### Get Markets
+Get all v2 Market objects
+
+```javascript
+const { markets } = await perpetual.api.getMarketsV2();
+```
```
Comment (BrendanChou) [inline docs/perpetual-typescript.md:109]: sorta unrelated but do want to change this to `getMarkets` and change it in the perp library as well?

Comment (samweinberg23) [reply docs/perpetual-typescript.md:109]: Sure - I can make a PR for that now

**Pull request merged**